src/lib/y2users/config.rb

```diff
@@ -199,5 +209,92 @@ def self.next
         @last_element_id += 1
       end
     end
+
+    # Helper class to merge users and groups of two configs
+    class Merger
+      # Constructor
+      #
+      # @param lhs [Config] Left Hand Size config
+      # @param rhs [Config] Right Hand Size config
+      def initialize(lhs, rhs)
+        @lhs = lhs
+        @rhs = rhs
+      end
+
+      def merge
+        merge_users
+        merge_groups
+      end
+
+    private
+
+      # @return [Config] Left Hand Size config
+      attr_reader :lhs
+
+      # @return [Config] Right Hand Size config
+      attr_reader :rhs
+
+      def merge_users
+        rhs.users { |u| merge_user(u) }
+      end
+
+      def merge_groups
+        rhs.groups { |g| merge_group(g) }
+      end
+
+      def merge_user(rhs_user)
+        lhs_user = lhs.users.find { |u| u.name == rhs_user.name }
+
+        lhs_user ? merge_existing_user(lhs_user, rhs_user) : merge_new_user(rhs_user)
+      end
+
+      def merge_new_user(rhs_user)
+        user = rhs_user.clone
+
+        lhs.attach(user)
+      end
+
+      def merge_existing_user(lhs_user, rhs_user)
+        user = rhs_user.clone
+
+        user.assign_internal_id(lhs_user.id)
+
+        copy_user_values(user, lhs_user)
+
+        lhs.detach(lhs_user)
+        lhs.attach(user)
+      end
+
+      def copy_user_values(user, lhs_user)
+        nil
+      end
+
+      def merge_group(rhs_group)
+        lhs_group = lhs.groups.find { |u| u.name == rhs_group.name }
+
+        lhs_group ? merge_existing_group(lhs_group, rhs_group) : merge_new_group(rhs_group)
+      end
+
+      def merge_new_group(rhs_group)
+        group = rhs_group.clone
+
+        lhs.attach(group)
+      end
+
+      def merge_existing_group(lhs_group, rhs_group)
+        group = rhs_group.clone
+
+        group.assign_internal_id(lhs_group.id)
```

Comment:
I am not sure if same name always means just edit. What is reason for this?

Reply:
Ok, I think that copying the id should not be the default for `ConfigMergers::Replace`. But, most likely we will need to preserve the id when we are merging users_simple config into the system config. If so, we can add a new `ConfigMergers::UsersSimple`.